node/task.ts

```diff
@@ -659,6 +665,22 @@ export function getPlatform(): Platform {
     }
 }
 
+/**
+ * Return hosted type of Agent
+ * @returns {AgentHostedMode}
+ */
+export function getAgentMode(): AgentHostedMode {
+    let agentCloudId = getVariable('Agent.CloudId');
+
+    if (agentCloudId === undefined)
```

Comment:
It would be a good idea, but here is a little bit of another logic. "undefined" value means that the current agent version doesn't support Agent.CloudId, it's for backward compatibility. Any other empty values mean it's SelfHosted and real value means MsHosted